### PR TITLE
update templates dependencies

### PR DESCRIPTION
**Pull request by cheahuychou**
Update the dependencies of all template services and migrate to cucumber 2

### CODE REVIEW

File: exosphere-shared/templates/add-service/exoservice-es6-mongodb/features/steps/steps.js [hunk 1]
```diff
@@ -43,14 +44,14 @@ module.exports = function() {
...
+  When(/^receiving the message "([^"]*)"$/, function(message) {
     this.exocom.send({ service: '_____serviceRole_____',
                        name: message })
   })
 
 
-  this.When(/^receiving the message "([^"]*)" with the payload:$/, function(message, payload, done) {
-    this.fillIn_____modelName@camelcase_____Ids(payload, (filledPayload) => {
+  When(/^receiving the message "([^"]*)" with the payload:$/, function(message, payload, done) {
+    this.fillIn_____modelNameCamelcase_____Ids(payload, (filledPayload) => {
```
Comment (charlierudolph) [inline exosphere-shared/templates/add-service/exoservice-es6-mongodb/features/steps/steps.js]: was this change intentional? I don't see why it would have changed. This change occurs quite a few times as well

Comment (hugoesthere) [inline exosphere-shared/templates/add-service/exoservice-es6-mongodb/features/steps/steps.js]: This is what [tmplconv](https://github.com/okunishinishi/node-tmplconv) uses to render templates right? Why was this changed?

File: exosphere-shared/templates/add-service/exoservice-ls-mongodb/features/support/world.ls [hunk 2]
```diff
@@ -6,7 +6,7 @@ require! {
 World = !->
 
   # Fills in _____modelName_____ ids in the placeholders of the template
-  @fill-in-_____modelName_____-ids = (template, done) ->
+  @fill-in-modelName-ids = (template, done) ->
```
Comment (charlierudolph) [inline exosphere-shared/templates/add-service/exoservice-ls-mongodb/features/support/world.ls]: Was this change intentional? Along the same vain as the previous comment

File: exosphere-shared/templates/add-service/exoservice-es6-mongodb/features/steps/steps.js (hunk 1 above)
Comment (cheahuychou) [reply exosphere-shared/templates/add-service/exoservice-es6-mongodb/features/steps/steps.js]: It was intentional. The original (master) one returns this error
```    
this.fillIn_____modelName@camelcase_____Ids(payload, (filledPayload) => {
                             ^
SyntaxError: Invalid or unexpected token
```

File: exosphere-shared/templates/add-service/exoservice-ls-mongodb/features/support/world.ls (hunk 2 above)
Comment (cheahuychou) [reply exosphere-shared/templates/add-service/exoservice-ls-mongodb/features/support/world.ls]: This was also intentional. Keeping the original name results in this error:
```
ReferenceError: fillIn is not defined
         at World.<anonymous>
```
and in master version:
```
ReferenceError: _____modelName_____Ids is not defined
         at World.<anonymous>
```

File: exosphere-shared/templates/add-service/exoservice-es6-mongodb/features/steps/steps.js (hunk 1 above)
Comment (cheahuychou) [reply exosphere-shared/templates/add-service/exoservice-es6-mongodb/features/steps/steps.js]: This function is defined in `support/world.js`. I might be wrong but I think the logic that involves `tmplcov` is there. I only changed the function name to fix the syntax error. The issue is that the feature tests for `exoservice-es6-mongodb` and `exoservice-ls-mongodb` both failed originally in the master version.

File: exosphere-shared/templates/add-service/exoservice-ls-mongodb/features/support/world.ls (hunk 2 above)
Comment (charlierudolph) [reply exosphere-shared/templates/add-service/exoservice-ls-mongodb/features/support/world.ls]: You shouldn't be running the tests of these folders without those lines being replaced. They are placeholders that get converted when a service gets scaffolded

File: exosphere-shared/templates/add-service/exoservice-es6-mongodb/features/steps/steps.js (hunk 1 above)
Comment (charlierudolph) [reply exosphere-shared/templates/add-service/exoservice-es6-mongodb/features/steps/steps.js]: Thats fine, we care only if it errors after those placeholders have been filled in during the process of creating a service

File: exosphere-shared/templates/add-service/exoservice-ls-mongodb/features/support/world.ls (hunk 2 above)
Comment (cheahuychou) [reply exosphere-shared/templates/add-service/exoservice-ls-mongodb/features/support/world.ls]: Sorry about that. I didn't know I wasn't supposed to run the tests. I've undone those changes.

Comment (hugoesthere) [reply exosphere-shared/templates/add-service/exoservice-ls-mongodb/features/support/world.ls]: You should run the tests after the placeholders have been filled in (i.e. if you were to run `exo-add` with one of these services). Do the tests pass after doing that?

Comment (cheahuychou) [reply exosphere-shared/templates/add-service/exoservice-ls-mongodb/features/support/world.ls]: Yes.

File: exosphere-shared/templates/add-service/exoservice-ls-mongodb/features/steps/steps.ls [hunk 3]
```diff
@@ -50,24 +50,24 @@ module.exports = ->
...
+  Then /^the service now contains the _____modelName_____s:$/, (table, done) ->
     @exocom
       ..send service: '_____serviceRole_____', name: '_____modelName_____.list'
       ..on-receive ~>
-        actual-_____modelName_____s = @remove-ids @exocom.received-messages[0].payload
-        expected-_____modelName_____s = [lowercase-keys(_____modelName_____) for _____modelName_____ in table.hashes!]
-        jsdiff-console actual-_____modelName_____s, expected-_____modelName_____s, done
+        actual_____modelName_____s = @remove-ids @exocom.received-messages[0].payload
+        expected_____modelName_____s = [lowercase-keys(_____modelName_____) for _____modelName_____ in table.hashes!]
+        jsdiff-console actual_____modelName_____s, expected_____modelName_____s, done
```
Comment (charlierudolph) [inline exosphere-shared/templates/add-service/exoservice-ls-mongodb/features/steps/steps.ls]: I believe the changes to these lines should be reverted

**Review by charlierudolph — APPROVED** [review]
Good to merge after comment is addressed
